kahypar/partition/coarsening/ml_coarsener.h

```diff
@@ -140,6 +166,9 @@ class MLCoarsener final : public ICoarsener,
         }
       }
     }
+    if (_tmp_ratings.contains(u)) {
+        _tmp_ratings.remove(u);
+    }
```

Comment:
This if seems to be unnecessary, since you explicitly check for v!=u in the loop above.